src/ExpenseTrackerGrupo2/DataAccess/Layer/BaseRepository.cs

```diff
@@ -0,0 +1,47 @@
+using Dapper;
+using ExpenseTrackerGrupo2.src.ExpenseTrackerGrupo2.Data;
+
+namespace ExpenseTrackerGrupo2.src.ExpenseTrackerGrupo2.DataAccess.Layer;
+public class BaseRepository<T> : IBaseRepository<T> where T : class
+{
+    protected readonly IDbConnectionFactory _dbConnectionFactory;
+
+    public BaseRepository(IDbConnectionFactory IDbConnectionFactory)
+    {
+        _dbConnectionFactory = IDbConnectionFactory;
+    }
+
+    public async Task<IList<T>> GetAll()
+    {
+        var connection = await _dbConnectionFactory.CreateConnection();
+        return (await connection.QueryAsync<T>("SELECT * FROM " + typeof(T).Name)).ToList();
+    }
+
+    public async Task<T> GetById(Guid id)
+    {
+        var connection = await _dbConnectionFactory.CreateConnection();
+        var result = await connection.QuerySingleOrDefaultAsync<T>("SELECT * FROM " + typeof(T).Name + " WHERE Id = @Id", new { Id = id });
```

Comment:
Please don't forget to apply database naming convention inside the query. You can review **PR #18** to see how the tables and columns name are defined.

Reply:
Also, review the rest of repositories and its queries according on what I mention before.

Reply:
Done